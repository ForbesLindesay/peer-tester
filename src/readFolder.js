import lsr from 'lsr';
import {readFile} from 'then-fs';

async function readFolder(dirname) {
  const entries = await lsr(dirname);
  return Promise.all(
    entries.map(async (entry) => {
      if (entry.isFile()) {
        return {
          type: 'file',
          path: entry.path,
          content: await readFile(entry.fullPath),
        };
      } else {
        return {
          type: 'directory',
          path: entry.path,
        };
      }
    }),
  );
}

export default readFolder;
