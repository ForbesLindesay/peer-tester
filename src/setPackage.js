function setPackage(entries, pkg) {
  return entries.map(entry => {
    if (entry.path === './package.json') {
      return {
        ...entry,
        content: new Buffer(JSON.stringify(pkg, null, '  ')),
      };
    }
    return entry;
  })
}
export default setPackage;
