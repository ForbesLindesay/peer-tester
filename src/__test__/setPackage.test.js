import setPackage from '../setPackage';
import readFolder from '../readFolder';

test('setPackage(readFolder(exampleDir), {"hello": "world"})', async () => {
  const entries = await readFolder(__dirname + '/../../example');
  const entriesWithPkg = setPackage(entries, {"hello": "world"});
  expect(entriesWithPkg).toMatchSnapshot();
});
