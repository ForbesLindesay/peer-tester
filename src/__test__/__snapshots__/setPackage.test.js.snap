// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setPackage(readFolder(exampleDir), {"hello": "world"}) 1`] = `
Array [
  Object {
    "content": Buffer("{
  \\"hello\\": \\"world\\"
}"),
    "path": "./package.json",
    "type": "file",
  },
  Object {
    "path": "./test",
    "type": "directory",
  },
  Object {
    "content": Buffer("console.log('promise version v' + require('promise/package.json').version);
if (require('promise/package.json').version === '6.0.1') {
  console.log('Simulating faield test for v6.0.1');
  process.exit(1);
}
"),
    "path": "./test/index.js",
    "type": "file",
  },
]
`;
